tests/e2e/lib/apps.go

```diff
@@ -449,105 +445,138 @@ func RunMustGather(oc_cli string, artifact_dir string) error {
 	return err
 }
 
-func makeRequest(request string, api string, todo string) {
-	params := url.Values{}
-	params.Add("description", todo)
-	body := strings.NewReader(params.Encode())
-	req, err := http.NewRequest(request, api, body)
-	if err != nil {
-		log.Printf("Error making post request to todo app before Prebackup  %s", err)
-	}
-	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		log.Printf("Response of todo POST REQUEST  %s", resp.Status)
-	}
-	defer resp.Body.Close()
-}
-
 // VerifyBackupRestoreData verifies if app ready before backup and after restore to compare data.
-func VerifyBackupRestoreData(clientv1 client.Client, artifact_dir string, namespace string, routeName string, app string, prebackupState bool, twoVol bool, backupRestoretype BackupRestoreType) error {
+func VerifyBackupRestoreData(ocClient client.Client, kubeClient *kubernetes.Clientset, kubeConfig *rest.Config, artifactDir string, namespace string, routeName string, serviceName string, app string, prebackupState bool, twoVol bool) error {
 	log.Printf("Verifying backup/restore data of %s", app)
-	appRoute := &routev1.Route{}
-	backupFile := artifact_dir + "/backup-data.txt"
-	routev1.AddToScheme(clientv1.Scheme())
-	err := clientv1.Get(context.Background(), client.ObjectKey{
-		Namespace: namespace,
-		Name:      routeName,
-	}, appRoute)
+	appEndpointURL, proxyPodParams, err := getAppEndpointURLAndProxyParams(ocClient, kubeClient, kubeConfig, namespace, serviceName, routeName)
 	if err != nil {
 		return err
 	}
-	appApi := "http://" + appRoute.Spec.Host
-	appEndpoint := appApi + "/todo-incomplete"
-	volumeEndpoint := appApi + "/log"
 
-	//if this is prebackstate = true, add items via makeRequest function. We only want to make request before backup
-	//and ignore post restore checks.
-	log.Printf("PrebackState: %t\n", prebackupState)
+	// Construct request parameters for the "todo-incomplete" endpoint
+	requestParamsTodoIncomplete := getRequestParameters(appEndpointURL+"/todo-incomplete", proxyPodParams, GET, nil)
+
 	if prebackupState {
-		// delete backupFile if it exists
-		if _, err := os.Stat(backupFile); err == nil {
-			os.Remove(backupFile)
-		}
-		//data before curl request
-		dataBeforeCurl, err := getResponseData(appEndpoint)
+		// Clean up existing backup file
+		RemoveFileIfExists(artifactDir + "/backup-data.txt")
+
+		// Make requests and update data before backup
+		dataBeforeCurl, _, err := MakeRequest(*requestParamsTodoIncomplete)
```

Comment:
All places that call `MakeRequest` (to check it `grep -Inr 'MakeRequest' tests/e2e`) ignore second return variable. Use it or remove from function

Reply:
Ok I've made that change. We are interested in that message and separate this from other errors, because we need to know if it was transport or for example api issue.